modules/ting_field_search/ting_field_search.module

```diff
@@ -432,39 +433,202 @@ function ting_field_search_ting_search_settings_submit($form, &$form_state) {
 }
 
 /**
- * Implements hook_form_FORM_ID_alter().
+ * Implements hook_preprocess_html().
+ *
+ * Enable the search box to be expanded based on the path of the profiles.
  */
-function ting_field_search_form_search_block_form_alter(&$form, &$form_state) {
-  if (!variable_get('ting_field_search_search_style', FALSE)) {
-    return;
+function ting_field_search_preprocess_html(&$vars) {
+  $profiles = ting_field_search_profiles_load_all();
+
+  foreach ($profiles as $profile) {
+    // Expended search box not checked go to the next profile.
+    if (!array_key_exists('search_box', $profile->config['user_interaction']) || !$profile->config['user_interaction']['search_box']) {
```

Comment:
Could this be replaced with 

```php
if (!empty($profile->config['user_interaction']['search_box'])) {
```

Reply:
It have been changed from your suggestion, based on internal code review to the current form.